### ISSUE TITLE
runner.Run() in TF# runs 5-10 times slower than session.Run(***) in TF-python

### ISSUE THREAD

**Issue by ys10** [migueldeicaza/TensorFlowSharp#445]
**Describe**
I implemented a **Griffin-Lim algorithm**(GL, a speech processing method used for reconstructing wave from mel-spectrogram) with TF-python, and dump it into a pb file.
Then I load graph of Griffin-Lim from the pb file, and using runner.Run() to call the computing graph and get the output of the last Node.

The result of output is correct and the voice quality is on par with TF-python generated.
But runner.Run() in TF# runs 5-10 times slower than session.Run(***) in TF-python.
For example, to generate a **3 seconds wave**, runner.Run() in TF# need about **2.7 seconds**, but session.Run(***) in TF-python only need **0.3-0.6 seconds**.

here are the generate key codes:

                TFTensor input = new TFTensor(melSpectrum);
                var runner = this.session.GetRunner();
                runner.AddInput(this.input, input);
                runner.Fetch(this.output);
                TFTensor[] results = **runner.Run()**;

**To Reproduce**
I went back and checked the related code I call runner.Run().
1. I have already kept a TFSession instant, so the solution in https://github.com/migueldeicaza/TensorFlowSharp/issues/330 can not help.
2.When I initialize the session & graph, I also warm up them by faking a melSpectrum and calling upper codes. If sess& graph weren't warmed up, runner.Run() will cost 6.8-7seconds to generate a 3 seconds wav.

**Expected behavior**
I think runner.Run()  in TF# should run as fast as session.Run() in TF-python

Any clues?
Thanks

**Comment by ys10** [comment]
Last line of key codes should be 
            TFTensor[] results = runner.Run();
No star character was in use.

**Comment by mg021** [comment]
@ys10 
I have same problem any help ???

**Comment by malibu** [comment]
You're right #330 can not solve the issue if you're not creating new sessions. I've seen this sort of disparity between python and .NET C++ wrappers like this library for a variety of reasons. I'll list just write down few that have caused me headaches in comparing the languages subsystems for reference. 

- Version differences between underlying libraries. Check Tensorflow versions used (libtensorflow.dll) between python and Tensorflow sharp.

- GPU usage in python code. Check if there is any GPU usage in python vs C# code. I assume you're running both on CPU or both on GPU on the same machine. Check CPU and GPU utilization to see what is being engaged when you run the session

- Bigger graphs in tensorflowsharp can cause marshalling time as managed wrapper spends time marshalling data between managed stack and unmanaged stack. Check the performance for a smaller, well known sample like an image object detection.

**Comment by faridi94** [comment]
@ys10 
i'm also having the same issue. Did you solve it by any chance?
